test/provenance.cpp

```diff
@@ -0,0 +1,297 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "mock_provenance_config.hpp"
+#include "ngraph/ngraph.hpp"
+
+using namespace std;
+using namespace ngraph;
+using ::testing::Return;
+
+using ProvSet = std::unordered_set<std::string>;
+
+TEST(provenance, provenance)
+{
+    auto provenance_config = std::make_shared<MockProvenanceConfig>();
+    EXPECT_CALL(*provenance_config, is_enabled()).WillRepeatedly(Return(true));
```

Comment:
Instead of using mocks here, let's just pass `true` to our `replace_node` calls.

Reply:
That's be definitely simpler. But I believe, @diyessi would have an objection to it. He is suggesting to use either pass_config.hpp or global getter-setter. I think we need a offline quick discussion.

Reply:
I'd actually prefer a global getter/setter as well. (Sorry, should have updated this comment.)

Reply:
I'd prefer the pass one was global, so global is okay here.